src/gt4py/next/program_processors/runners/dace_fieldview/gtir_python_codegen.py

```diff
@@ -0,0 +1,128 @@
+    "arctanh": "atanh({})",
+    "sqrt": "math.sqrt({})",
+    "exp": "math.exp({})",
+    "log": "math.log({})",
+    "gamma": "tgamma({})",
+    "cbrt": "cbrt({})",
+    "isfinite": "isfinite({})",
+    "isinf": "isinf({})",
+    "isnan": "isnan({})",
+    "floor": "math.ifloor({})",
+    "ceil": "ceil({})",
+    "trunc": "trunc({})",
+    "minimum": "min({}, {})",
+    "maximum": "max({}, {})",
+    "fmod": "fmod({}, {})",
+    "power": "math.pow({}, {})",
+    "float": "dace.float64({})",
+    "float32": "dace.float32({})",
+    "float64": "dace.float64({})",
+    "int": "dace.int32({})" if np.dtype(int).itemsize == 4 else "dace.int64({})",
+    "int32": "dace.int32({})",
+    "int64": "dace.int64({})",
+    "bool": "dace.bool_({})",
+    "plus": "({} + {})",
+    "minus": "({} - {})",
+    "multiplies": "({} * {})",
+    "divides": "({} / {})",
+    "floordiv": "({} // {})",
+    "eq": "({} == {})",
+    "not_eq": "({} != {})",
+    "less": "({} < {})",
+    "less_equal": "({} <= {})",
+    "greater": "({} > {})",
+    "greater_equal": "({} >= {})",
+    "and_": "({} & {})",
+    "or_": "({} | {})",
+    "xor_": "({} ^ {})",
+    "mod": "({} % {})",
+    "not_": "(not {})",  # ~ is not bitwise in numpy
+}
+
+
+def format_builtin(bultin: str, *args: Any) -> str:
+    if bultin in MATH_BUILTINS_MAPPING:
+        fmt = MATH_BUILTINS_MAPPING[bultin]
+    else:
+        raise NotImplementedError(f"'{bultin}' not implemented.")
+    return fmt.format(*args)
+
+
+class PythonCodegen(codegen.TemplatedGenerator):
+    """Helper class to visit a symbolic expression and translate it to Python code.
+
+    The generated Python code can be use either as the body of a tasklet node or,
+    as in the case of field domain definitions, for sybolic array shape and map range.
+    """
+
+    SymRef = as_fmt("{id}")
+    Literal = as_fmt("{value}")
+
+    def _visit_deref(self, node: gtir.FunCall) -> str:
+        assert len(node.args) == 1
+        if isinstance(node.args[0], gtir.SymRef):
+            return self.visit(node.args[0])
+        raise NotImplementedError(f"Unexpected deref with arg type '{type(node.args[0])}'.")
+
+    def _visit_numeric_builtin(self, node: gtir.FunCall) -> str:
+        assert isinstance(node.fun, gtir.SymRef)
+        fmt = MATH_BUILTINS_MAPPING[str(node.fun.id)]
+        args = self.visit(node.args)
+        return fmt.format(*args)
+
+    def visit_FunCall(self, node: gtir.FunCall) -> str:
+        if cpm.is_call_to(node, "deref"):
+            return self._visit_deref(node)
+        elif isinstance(node.fun, gtir.SymRef):
+            args = self.visit(node.args)
+            builtin_name = str(node.fun.id)
+            return format_builtin(builtin_name, *args)
+        raise NotImplementedError(f"Unexpected 'FunCall' node ({node}).")
+
+
+"""
+Specialized visit method for symbolic expressions.
+
+Returns:
+    A string containing the Python code corresponding to a symbolic expression
+"""
+get_source = PythonCodegen.apply
```

Comment:
It's not intuitive, but the docstring goes after the definition not before
```suggestion
get_source = PythonCodegen.apply
"""
Specialized visit method for symbolic expressions.

Returns:
    A string containing the Python code corresponding to a symbolic expression
"""
```